mava/wrappers/gigastep.py

```diff
@@ -0,0 +1,332 @@
+        action_spec: Returns the action spec for the environment.
+        reward_spec: Returns the reward spec for the environment.
+        discount_spec: Returns the discount spec for the environment.
+        _split_obs_and_state: Splits the observation and state into separate arrays for each team.
+        get_wining_team: Returns the winning team based on the number of alive agents.
+        adversary_policy: Chooses an action for the adversary.
+    """
+
+    def __init__(
+        self,
+        env: GigastepEnv,
+        time_limit: int = 500,
+        has_global_state: bool = False,
+        get_adversary_obs: bool = False,
+        get_adversary_actions: bool = False,
+    ):
+
+        assert (
+            env.discrete_actions
+        ), "Only discrete action spaces are currently supported for Gigastep environments"
+        assert (
+            env._obs_type == "vector"
+        ), "Only Vector observations are currently supported for Gigastep environments"
+        assert (
+            get_adversary_actions is False and get_adversary_obs is False
+        ) or has_global_state is True, (
+            "For a customized global observation, set has_global_state to True"
+        )
+
+        super().__init__(env)
+        self._env: GigastepEnv
+        self._env.max_episode_length = time_limit
+        self.num_agents = self._env.n_agents_team1
+        self.num_actions = self._env.n_actions
+
+        self.has_global_state = has_global_state
+        self.get_adversary_obs = get_adversary_obs
+        self.get_adversary_actions = get_adversary_actions
+
+    def reset(self, key: PRNGKey) -> Tuple[GigasteplState, TimeStep]:
+        """
+        Reset the GIGASTEP environment.
+
+        Args:
+            key (PRNGKey): The PRNGKey.
+
+        Returns:
+            GigasteplState : the state of the environment.
+            TimeStep : the first time step.
+
+        """
+
+        key, reset_key, adversary_key = jax.random.split(key, 3)
+        obs, state = self._env.reset(reset_key)
+
+        team1_obs, team1_state, team2_obs, team2_state = self._split_obs_and_state(obs, state)
+
+        adversary_actions = self.adversary_policy(team2_obs, team2_state, adversary_key)
+
+        if self.has_global_state:
+            obs = ObservationGlobalState(
+                agents_view=team1_obs,
+                action_mask=self.action_mask(),
+                global_state=self.get_global_state(obs, adversary_actions),
+                step_count=jnp.zeros(self.num_agents, dtype=int),
+            )
+        else:
+            obs = Observation(
+                agents_view=team1_obs,
+                action_mask=self.action_mask(),
+                step_count=jnp.zeros(self.num_agents, dtype=int),
+            )
+
+        return GigasteplState(state, key, 0, adversary_actions), restart(
+            obs, shape=(self.num_agents,), extras={"won_episode": jnp.nan}
+        )
```

Comment:
2 things here, 
* slightly cleaner to create these in variables and then return them
* Can we make the `won_episode` always a bool


```suggestion
        state = GigasteplState(state, key, 0, adversary_actions)
        timestep = restart(obs, shape=(self.num_agents,), extras={"won_episode": False})
        return state, timestep
```